### ISSUE TITLE
System.Net.WebException: Error: NameResolutionFailure

### ISSUE THREAD

**Issue by JakubHolovsky** [xamarin/Xamarin.Auth#187]
# Xamarin.Auth Issue

### Version

- [ ] nuget version = 1.5.0
- [ ] component version = 
- [ ] sample

### Steps to reproduce

1. Can't reproduce (it's coming from our crash analytics), one off bug but might lead you to potentially fixing the issue.

> Android: 5.1.1
> Android Build: LMY48B.J500GXXU1AOL1
> Manufacturer: samsung
> Model: SM-J500G
> Thread: Thread-2666-2666
> CrashReporter Key: A6142E57-3598-E073-FC61-BF4DDEFDCB44BD56B087
> Start Date: Mon Jun 05 23:56:02 GMT+08:00 2017
> Date: Tue Jun 06 00:04:28 GMT+08:00 2017
> Format: Xamarin
> 
> Xamarin caused by: android.runtime.JavaProxyThrowable: System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. ---> System.AggregateException: One or more errors occurred. ---> System.Net.WebException: Error: NameResolutionFailure
>   at System.Net.HttpWebRequest.EndGetResponse (System.IAsyncResult asyncResult) [0x00058] in <e9eb39db51e4425891101ef19f46bc76>:0 
>   at System.Threading.Tasks.TaskFactory`1[TResult].FromAsyncCoreLogic (System.IAsyncResult iar, System.Func`2[T,TResult] endFunction, System.Action`1[T] endAction, System.Threading.Tasks.Task`1[TResult] promise, System.Boolean requiresSynchronization) [0x0000f] in <a83cd6d69e484f55bdb92ebf1dda2881>:0 
>    --- End of inner exception stack trace ---
>   at System.Threading.Tasks.Task.ThrowIfExceptional (System.Boolean includeTaskCanceledExceptions) [0x00011] in <a83cd6d69e484f55bdb92ebf1dda2881>:0 
>   at System.Threading.Tasks.Task`1[TResult].GetResultCore (System.Boolean waitCompletionNotification) [0x0002b] in <a83cd6d69e484f55bdb92ebf1dda2881>:0 
>   at System.Threading.Tasks.Task`1[TResult].get_Result () [0x0000f] in <a83cd6d69e484f55bdb92ebf1dda2881>:0 
>   at Xamarin.Auth.OAuth1Authenticator.<GetInitialUrlAsync>m__0 (System.Threading.Tasks.Task`1[TResult] respTask) [0x00000] in <bae572ba45754aac977da65c58a52c3b>:0 
>   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2[TAntecedentResult,TResult].InnerInvoke () [0x00024] in <a83cd6d69e484f55bdb92ebf1dda2881>:0 
>   at System.Threading.Tasks.Task.Execute () [0x00010] in <a83cd6d69e484f55bdb92ebf1dda2881>:0 
>    --- End of inner exception stack trace ---
> ---> (Inner Exception #0) System.AggregateException: One or more errors occurred. ---> System.Net.WebException: Error: NameResolutionFailure
>   at System.Net.HttpWebRequest.EndGetResponse (System.IAsyncResult asyncResult) [0x00058] in <e9eb39db51e4425891101ef19f46bc76>:0 
>   at System.Threading.Tasks.TaskFactory`1[TResult].FromAsyncCoreLogic (System.IAsyncResult iar, System.Func`2[T,TResult] endFunction, System.Action`1[T] endAction, System.Threading.Tasks.Task`1[TResult] promise, System.Boolean requiresSynchronization) [0x0000f] in <a83cd6d69e484f55bdb92ebf1dda2881>:0 
>    --- End of inner exception stack trace ---
>   at System.Threading.Tasks.Task.ThrowIfExceptional (System.Boolean includeTaskCanceledExceptions) [0x00011] in <a83cd6d69e484f55bdb92ebf1dda2881>:0 
>   at System.Threading.Tasks.Task`1[TResult].GetResultCore (System.Boolean waitCompletionNotification) [0x0002b] in <a83cd6d69e484f55bdb92ebf1dda2881>:0 
>   at System.Threading.Tasks.Task`1[TResult].get_Result () [0x0000f] in <a83cd6d69e484f55bdb92ebf1dda2881>:0 
>   at Xamarin.Auth.OAuth1Authenticator.<GetInitialUrlAsync>m__0 (System.Threading.Tasks.Task`1[TResult] respTask) [0x00000] in <bae572ba45754aac977da65c58a52c3b>:0 
>   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2[TAntecedentResult,TResult].InnerInvoke () [0x00024] in <a83cd6d69e484f55bdb92ebf1dda2881>:0 
>   at System.Threading.Tasks.Task.Execute () [0x00010] in <a83cd6d69e484f55bdb92ebf1dda2881>:0 
> ---> (Inner Exception #0) System.Net.WebException: Error: NameResolutionFailure
>   at System.Net.HttpWebRequest.EndGetResponse (System.IAsyncResult asyncResult) [0x00058] in <e9eb39db51e4425891101ef19f46bc76>:0 
>   at System.Threading.Tasks.TaskFactory`1[TResult].FromAsyncCoreLogic (System.IAsyncResult iar, System.Func`2[T,TResult] endFunction, System.Action`1[T] endAction, System.Threading.Tasks.Task`1[TResult] promise, System.Boolean requiresSynchronization) [0x0000f] in <a83cd6d69e484f55bdb92ebf1dda2881>:0 <---
> <---
> 

Platform:

-	[] .NET version = 
-	[] mono version = 5.0.1.1

### Expected behaviour

The app should not crash if "resolution failure" happens.

**Comment by moljac** [comment]
This is now design problem.

OK. I might put it in Xamarin.Auth, but how do you want me to inform user that they have DNS problems (possible network problems).

Everything in Xamarin.Auth starts (and finishes) from ctor. I would put ctor in try/catch and then decide to log or show to user.

But thanks for investigating and reporting

**Comment by JakubHolovsky** [comment]
@moljac shouldn't this trigger error event? I can put the try catch to my app but just wondering what the best approach is here.

**Comment by moljac** [comment]
Xamarin.Auth does not handle networking issues/interruptions during OAuth flow.

Due to lack of info and difficulty of reproducing triaging it as "Low Priority"
